robottelo/performance/thread.py

```diff
@@ -60,11 +60,17 @@ def run(self):
 
 
 class SubscribeAttachThread(PerformanceThread):
-    def __init__(self, thread_id, thread_name, time_result_dict,
-                 num_iterations, sub_id, default_org, environment, vm_ip):
+    def __init__(self, thread_id, thread_name,
```

Comment:
Method `PerformanceThread.__init__` has the following signature:

```python
    def __init__(self, thread_id, thread_name, time_result_dict):
```

But this method has the following signature:

```python
def __init__(self, thread_id, thread_name, time_result_dict_register, ...)
```

The subclass does not include `time_result_dict` as a fourth parameter, just like the parent class. It's legal, but very bothersome design. It's even more awkward that this is the set of arguments passed to `super`:

```python
super(SubscribeAttachThread, self).__init__(thread_id, thread_name, {})
```

So the subclass is *hiding* functionality provided by the parent class. This is a clear violation of the liskov violation principle!

If one class need to make use of another, it should either:

* properly subclass and adhere to the liskov substitution principle, or
* use composition and simply use `PerformanceThread` from within `SubscribeAttachThread`.